apps/coscrad-frontend/src/components/higher-order-components/categorizable-page-layout.tsx

```diff
@@ -0,0 +1,174 @@
+import { AggregateCompositeIdentifier } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    Close as CloseIcon,
+    ExpandLess as ExpandLessIcon,
+    ExpandMore as ExpandMoreIcon,
+    Hub as HubIcon,
+    TextSnippet as TextSnippetIcon,
+} from '@mui/icons-material';
+import {
+    Box,
+    Divider,
+    Drawer,
+    IconButton,
+    Stack,
+    Tooltip,
+    Typography,
+    styled,
+} from '@mui/material';
+import { ReactNode, useState } from 'react';
+import { CommandPanel } from '../commands';
+
+interface CategorizablePageLayoutProps {
+    compositeIdentifier: AggregateCompositeIdentifier;
+    selfNotesList: JSX.Element;
+    connectedResourcesList: JSX.Element;
+    commandPanel?: JSX.Element;
+    children: ReactNode;
+}
+
+const DrawerHeader = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'flex-start',
+    margin: theme.spacing(2, 4, 0, 4),
+    justifyContent: 'space-between',
+}));
+
+const DrawerContentStack = styled(Stack)(({ theme }) => ({
+    margin: theme.spacing(0, 4, 0, 4),
+    overflow: 'scroll',
+}));
+
+type Anchor = 'top' | 'left' | 'bottom' | 'right';
+
+export const CategorizablePageLayout = ({
+    compositeIdentifier: { type: resourceType, id },
+    selfNotesList,
+    connectedResourcesList,
+    commandPanel,
+    children,
+}: CategorizablePageLayoutProps) => {
+    const [drawerState, setDrawerState] = useState({
+        bottom: false,
```

Comment:
I'm not exactly sure how things have evolved since the last view. We previously only wanted one panel to be open at a time.